src/OpenApi/src/Services/OpenApiDocumentService.cs

```diff
@@ -73,18 +69,27 @@ public async Task<OpenApiDocument> GetOpenApiDocumentAsync(IServiceProvider scop
         var document = new OpenApiDocument
         {
             Info = GetOpenApiInfo(),
-            Paths = await GetOpenApiPathsAsync(capturedTags, scopedServiceProvider, operationTransformers, schemaTransformers, cancellationToken),
-            Servers = GetOpenApiServers(),
-            Tags = [.. capturedTags]
+            Servers = GetOpenApiServers()
         };
+        document.Paths = await GetOpenApiPathsAsync(document, scopedServiceProvider, operationTransformers, schemaTransformers, cancellationToken);
+        document.Tags = document.Tags?.Distinct(OpenApiTagComparer.Instance).ToList();
```

Comment:
I filed https://github.com/microsoft/OpenAPI.NET/issues/1983 to eventually make this a lot smoother.